0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/GenerateCommand.scala

```diff
@@ -119,7 +134,32 @@ case class GenerateCommand(
       outputPair.map(_._1),
       migrations,
       outputPair.flatMap(_._2),
-      warnings = schemaErrors ++ ddlErrors ++ ddlWarnings)
+      warnings = schemaVerWarning ++ schemaVerError ++ schemaErrors ++ ddlErrors ++ ddlWarnings)
+  }
+
+  /**
+    * Checks if given list of versions have a gap
+    *
+    * e.g. ["1-0-0", "1-0-1", "1-0-3"] miss "1-0-2"
+    *
+    * @param versions list of versions, as strings, from self-describing schemas
+    * @return whether versions have a gap or not
+    */
+  private[ctl] def existMissingSchemaVersion(versions: List[String]): Boolean = {
+    val numOfVersions = versions.length
+
+    if (numOfVersions == 1) return false
+
+    val versionsArr = versions.map(version => version.split('-').map(_.toInt))
```

Comment:
This could be `List[SchemaVer]`, but we casted it to unnecessary vague `List[String]` and now trying to get back structure of `List[SchemaVer]`.

There's a notion of total and partial functions. If you say that it accepts `List[String]` and returns `Boolean` then it means that it works for **any** list of strings. But it will fail (with runtime exceptions - `String#toInt`, `List#head`!) for way too many strings. You can avoid this by switching back to `List[SchemaVer]`.